### PR TITLE
[react-big-calendar] add a new generic for resource type

### PR DESCRIPTION
**Pull request by onlyann**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://intljusticemission.github.io/react-big-calendar/examples/index.html#prop-resources 

Current definition incorrectly uses the same generic type for the resource and the event.
This also fixes the definition of `onNavigate`.

Upgrading to Typescript 2.9 since the example should be using a generic component and that is not supported by Typescript 2.8.

### CODE REVIEW

**Review by RajeshKumarBoora — APPROVED** [review]

File: types/react-big-calendar/index.d.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@
 //                 Sebastian Silbermann <https://github.com/eps1lon>
 //                 Paul Potsides <https://github.com/strongpauly>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.8
+// TypeScript Version: 2.9
```
Comment (eps1lon) [inline types/react-big-calendar/index.d.ts]: Why do we need to bump to 2.9?

Comment (onlyann) [reply types/react-big-calendar/index.d.ts]: To use generic JSX elements as it is done in the test file.
This landed in TS 2.9 : https://github.com/Microsoft/TypeScript/pull/22415

Comment (onlyann) [reply types/react-big-calendar/index.d.ts]: @eps1lon 

What is the implication of bumping to TypeScript 2.9  other than to get the CI build to run the test under this version?
Will that prevent consumers of this package from using it with TS 2.8 ?

I can always change the syntax in the test file to satisfy TS 2.8 if preferred.

Comment (onlyann) [reply types/react-big-calendar/index.d.ts]: @eps1lon I have changed the syntax to satisfy TS 2.8. Are you happy with the PR ?

Comment (amcasey) [reply types/react-big-calendar/index.d.ts]: You seem to have addressed his feedback.

**Pull request merged**